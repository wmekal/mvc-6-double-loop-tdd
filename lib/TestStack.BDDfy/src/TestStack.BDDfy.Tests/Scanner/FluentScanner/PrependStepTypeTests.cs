using System.Runtime.CompilerServices;
using ApprovalTests;
using TestStack.BDDfy.Reporters;
using Xunit;

namespace TestStack.BDDfy.Tests.Scanner.FluentScanner
{
    public class PrependStepTypeTests
    {
        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void VerifyPrependStepTitles()
        {
            var story = this.Given(_ => GivenAStepWithGivenInIt())
                .Given(_ => AStepWithoutGivenInIt())
                .And(_ => AndGivenAStepWithAndGivenInIt())
                .And(_ => AStepWithoutGivenInIt())
                .But(_ => ButInStep())
                .But(_ => NothingInStep())
                .When(_ => WhenStuff())
                .When(_ => Stuff())
                .Then(_ => ThenWeAreWinning())
                .Then(_ => WeAreWinning())
                .BDDfy();

            var textReporter = new TextReporter();
            textReporter.Process(story);
            Approvals.Verify(textReporter.ToString());
        }

        private void GivenAStepWithGivenInIt()
        {
            
        }

        private void AndGivenAStepWithAndGivenInIt()
        {
            
        }

        private void AStepWithoutGivenInIt()
        {
            
        }

        private void ButInStep()
        {
            
        }

        private void NothingInStep()
        {
            
        }

        private void WhenStuff()
        {
            
        }

        private void Stuff()
        {
            
        }

        private void ThenWeAreWinning()
        {
            
        }

        private void WeAreWinning()
        {
            
        }
    }
}