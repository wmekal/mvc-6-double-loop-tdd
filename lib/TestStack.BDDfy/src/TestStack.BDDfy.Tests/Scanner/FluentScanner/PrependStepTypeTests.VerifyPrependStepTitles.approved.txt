
Scenario: Verify prepend step titles
	Given a step with given in it
	Given a step without given in it
	  And given a step with and given in it
	  And a step without given in it
	  But in step
	  But nothing in step
	When stuff
	When stuff
	Then we are winning
	Then we are winning

